### ISSUE TITLE
Document type declarations for MathML?

### ISSUE THREAD

**Issue by murata2makoto** [w3c/epub-specs#1370]
Do we need document type declarations for MathML? Since MathML is always embedded within an HTML content document, document type declarations for MathML are unusable. Do people agree?

This issue is to create § B. Allowed External Identifiers as proposed in #1368.

**Comment by mattgarrish** [comment]
This is a tough one. Technically you can embed MathML even without it being listed as a core media type, the more I think about it.

The `object` element falls back to the html inside it, for example, so you should be able to include mathml files without epubcheck complaining.

I don't believe mathml will always render this way, but it might be useful to allow in the interests of letting people do as the please.

**Comment by murata2makoto** [comment]
@matt-curtis 

Just checking.  Are you saying that MathML documents are allowed as spine items if fallback is specified?

**Comment by murata2makoto** [comment]
@mattgarrish 

Just checking. Are you saying that MathML documents are allowed as spine items if fallback is specified?

**Comment by mattgarrish** [comment]
> Are you saying that MathML documents are allowed as spine items if fallback is specified?

Sure, I suppose that's yet another way they could appear in EPUB. But you can also embed them within an XHTML document using the `object` tag so long as you have an xhtml equivalent as a fallback, something like this:

```
<object data="equation.xml" type="application/mathml+xml">
   <div>2+2=4</div>
</object>
```

That's an intrinsic fallback for a foreign media type, after all.

**Comment by murata2makoto** [comment]
@mattgarrish 

MathML in `object` is possible, but this does not mean that the DTD for MathML can be referenced from the HTML document having the MathML subtree.

**Comment by mattgarrish** [comment]
> this does not mean that the DTD for MathML can be referenced from the HTML document having the MathML subtree.

Right, but the xml file containing the equation can, and these rules apply to any xml file type. That's why we might want to allow this.

**Comment by murata2makoto** [comment]
> Right, but the xml file containing the equation can, and these rules apply to any xml file type. That's why we might want to allow this.

Yes, you are right. Such a MathML document is not a spine item.  

Do people expect that entities predefined by MathML are usable in such MathML documents?

**Comment by mattgarrish** [comment]
> Do people expect that entities predefined by MathML are usable in such MathML documents?

That I don't know, but I think we can remain consistent in disallowing them for use in EPUB, at least until the day comes when the HTML serialization is allowed.

I'd be curious if a browser would even render them, given they would be invalid if the math was embedded, but too late for me to test tonight.

**Comment by iherman** [comment]
But isn't it possible already today to include MathML as a data file in the EPUB instance (ie, not in the spine but listed in the manifest) and accessed, eg, via a javascript? The point is that MathML may be used to interchange mathematical formulae, not only for direct display. It is the same as, possibly, storing a CSV file or some other XML files.

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2020-11-06-epub#section3).

- `RESOLVED:  Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373`
<details><summary><i class="differentiate">View the transcript</i></summary>
Wendy Reid: we had resolutions at the F2F, and further discussions on github<br/>
… and came to a happy place<br/>
Matt Garrish: <a href="https://github.com/w3c/epub-specs/pull/1368">https://github.com/w3c/epub-specs/pull/1368</a><br/>
Matt Garrish: where we ended up was…<br/>
… we put in an allowance for a specific set of external identifiers that we have put in an appendix<br/>
… we have SVG and MathML that are allowed to be used in content docs or in separate files<br/>
… and we made a restriction against external entities in the internal DTD subset<br/>
… so it prevents some security issues but eases authoring<br/>
… so we’ll no longer force people to remove SVG DTDs from tool-generated files<br/>
… I’m hoping this is it :)<br/>
Ivan Herman: tech comment<br/>
… in fact, the changes are such that<br/>
… makes possible something that I’m not sure we really use<br/>
… I can define as part of an internal entity something that won’t go out to the network<br/>
… I’m not sure if this feature is in use<br/>
… formal comment<br/>
… there was a formal resolution on the previous version; this PR slightly changes that<br/>
… can we get a formal resolution to merge, and also close a bunch of issues which were examples of the problem?<br/>
Proposed resolution: Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373 (Wendy Reid)<br/>
Garth Conboy: +1<br/>
Matt Garrish: +1<br/>
Ivan Herman: +1<br/>
Charles LaPierre: +1<br/>
Matthew Chan: +1<br/>
Wendy Reid: +1<br/>
Brady Duga: +1<br/>
George Kerscher: +1<br/>
Laura Brady: +1<br/>
Bill Kasdorf: +1<br/>
Ben Schroeter: +1<br/>
Resolution #1: Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373<br/>
</details>
